cpp/src/arrow/acero/swiss_join_internal.h

```diff
@@ -604,6 +609,16 @@ class SwissTableForJoinBuild {
   MemoryPool* pool_;
   int64_t hardware_flags_;
 
+  // One per batch.
+  //
+  // Informations like hashes and partitions of each batch.
+  //
+  struct BatchState {
+    std::vector<uint32_t> hashes;
+    std::vector<uint16_t> prtn_ranges;
+    std::vector<uint16_t> prtn_row_ids;
```

Comment:
Are these one value per row?

Reply:
It might be worth adding comments as to that (the cardinality of each BatchState member).

Reply:
Sure. Comments added.